doc/tools/test-tool.md

```diff
@@ -0,0 +1,160 @@
+# `TestTool`
+
+Runs dart tests for the current project.
+
+## Usage
+
+> _This tool is included in the [`coreConfig`][core-config] and is runnable by
+> default via `ddev test`._
+
+```dart
+// tool/dart_dev/config.dart
+import 'package:dart_dev/dart_dev.dart';
+
+final config = {
+  'test': TestTool() // configure as necessary
+};
+```
+
+## `test` vs `build_runner test`
+
+Historically, `pub run test` has been the canonical way to run Dart tests.
+With the introduction of the [build system][build-system], there is now a second
+way to run tests. Projects that rely on builder outputs must run tests via
+`pub run build_runner test`.
+
+The `TestTool` will make this choice for you. If the current project has a
+dependency on `build_test`, it will run `pub run build_runner test`. Otherwise
+it will default to running `pub run test`.
+
+> It [appears][test-future] as though the long term goal is to integrate the
+> build system into the test runner so that `pub run test` is once again the
+> canonical way to run tests.
+
+## Default behavior
+
+By default this tool will run `pub run test`, unless there is a dependency on
+`build_test`, in which case it will run `pub run build_runner test`.
+
+## Running a subset of tests
+
+When developing it is common to want to run a targeted subset of tests. The test
+runner supports targeting tests by path(s), preset(s), or by matching
```

Comment:
This mentions specifying a path to a test file, but it's not obvious from `ddev help test` how you would do that. Maybe we should make that clearer.

Reply:
🤕 I hadn't read far enough. The answer is below the `ddev help test` output. So, ignore me!